### PR TITLE
Feature/bio

### PR DESCRIPTION
**Pull request by FelixPriori**
# In this PR
## App.js
* Changed the Axios's request to get the guest badges to take in the parameter's id instead of the currently logged in user's id since we're looking for the badges of a different user.
* Moved all the functions into a custom hook in `src/hooks/useAppData.js`
## GuestProfile.js
- Added a check to see if they have a bio. If not, it shows a message that the user does not have a bio yet.
- The page now shows a list of that user's quests (separated in own component).
- Changed the class names to better tailor the components to the page itself.
## GuestProfile.scss
- Fixed some centering issues.
- Made the page responsive.

### CODE REVIEW

File: client/src/components/App/App.js [hunk 1]
```diff
@@ -135,33 +139,32 @@ export default function App() {
...
+            setState((prevState) => {
+              return {
+                ...prevState,
+                userQuests,
+                villagers,
+                classesData,
+              };
+            })
+        )
+        .catch((e) => console.log(e));
```
Comment (cawel) [inline client/src/components/App/App.js:167]: Simpler?

```suggestion
      const userQuest = await axios.get("/quests").data;
      const villagers = await axios.get("/villagers").data;
      const classesData = await axios.get("/classes").data;
      setState((prevState) => {
              return {
                ...prevState,
                userQuests,
                villagers,
                classesData,
              };
            });
```

Comment (FelixPriori) [reply client/src/components/App/App.js:167]: NIce!

Comment (FelixPriori) [reply client/src/components/App/App.js]: This isn't working actually. Not sure why...

Comment (FelixPriori) [reply client/src/components/App/App.js]:
```
TypeError: Cannot read property 'length' of undefined
ClassTabs
src/components/ClassTabs/ClassTabs.js:25
  22 | });
  23 | 
  24 | return (
> 25 |   <Tabs className="classTabOutline rounded" defaultActiveKey="profile"
     | ^  26 |     id="uncontrolled-tab-example"
  27 |     activeKey={key}
  28 |     onSelect={(k) => setKey(k)}>
```

File: client/src/hooks/useAppData.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+              classesProgressData,
+              villagers,
+              badges,
+              questsByVillager,
+              questsByAdventurer,
+              adventurers,
+            });
+          }
+        )
+        .catch((err) => console.log(err));
```
[the remark this answers is not in the document]
Comment (FelixPriori) [reply client/src/hooks/useAppData.js:114]: Yeah, I couldn't figure it out, so I figured I would refactor in custom hook first before hacking at it. The size of the App.js was giving me vertigo lol.

**Pull request merged**